packages/cadl-ranch-specs/http/models/inheritance/main.cadl

```diff
@@ -1,60 +1,112 @@
+model Cat extends Pet {
+  age?: int32;
 }
 
-model DerivedFromBaseClassB extends BaseClass {
-  @doc("An example property on a derived type")
-  derivedClassBProperty: string;
+@doc("The third level model A in the non-discriminator three levels inheritance.")
+model Siamese extends Cat {
+  smart?: boolean;
 }
 
-@discriminator("discriminatorProperty")
-@doc("Example base class that has a discriminator property.")
-model BaseClassWithDiscriminator extends BaseClass {}
-// Note: discriminator property is not defined on parent
+@doc("Base model for three levels inheritance with discriminator.")
+@discriminator("kind")
+model Fish {
+  age?: int32;
+}
+
+@doc("The second level model in three levels inheritance with discriminator and defined a new discriminator.")
+@discriminator("sharktype")
+model Shark extends Fish {
+  kind: "shark";
+}
 
-model DerivedFromBaseClassWithDiscriminatorA extends BaseClassWithDiscriminator {
-  discriminatorProperty: "A";
+@doc("The second level model in three levels inheritance with discriminator.")
+model Salmon extends Fish {
+  kind: "salmon";
+  friends?: Fish[];
+  hate?: Record<Fish>;
+  partner?: Fish;
 }
 
-model DerivedFromBaseClassWithDiscriminatorB extends BaseClassWithDiscriminator {
-  discriminatorProperty: "B";
+@doc("The third level model A in the three levels inheritance with discriminators.")
+model SawShark extends Shark {
+  sharktype: "saw";
 }
 
-@doc("Illustrates case where a basic model has polymorphic properties.")
-model ModelWithPolymorphicProperty {
-  @doc("Example polymorphic type property.")
-  polymorphicProperty: BaseClassWithDiscriminator;
+@doc("The third level model B in the non-discriminator three levels inheritance with discriminators.")
+model GoblinShark extends Shark {
+  sharktype: "goblin";
 }
 
 @scenario
-@scenarioDoc("Generate, send, and receive round-trip inherited model.")
-@route("/model")
+@scenarioDoc("Generate and send model. The valid input value is {name: \"abc\", age: 32, smart: true}")
+@route("/valid")
+@post
+op sendInheritanceValid(@body input: Siamese): OkResponse | ErrorResponse;
+
+@scenario
+@scenarioDoc("Generate and receive model. The return value is {name: \"abc\", age: 32, smart: true}")
+@route("/valid")
+@get
+op getInheritanceValid(): Siamese | ErrorResponse;
+
+@scenario
+@scenarioDoc("Generate, send, and receive round-trip bottom model.")
+@route("/valid")
 @put
-op setValue(@body input: BaseClass): BaseClass | ErrorResponse;
+op setInheritanceValid(@body input: Siamese): Siamese | ErrorResponse;
 
 @scenario
-@scenarioDoc("Generate, send, and receive round-trip model with a polymorphic property.")
-@route("/property")
+@route("/invalid")
+@scenarioDoc("Get a basic model that is invalid for the local strong kind.")
+@get
+op getInheritanceInvalid(): Pet | ErrorResponse;
+
+@scenario
+@scenarioDoc("Generate, send, and receive round-trip model that optional property is not presented.")
+@route("/empty")
```

Comment:
I'm not convinced we need multiple dedicated tests for invalid and empty, @timotheeguerin thoughts?

Reply:
These abnormal cases are found in bodycomplex :), below are my understanding:
- The `invalid` is for testing what happen when service return an incorrect type of property value, for example: return an integer value for a string property.
- The `null` is for testing when service return a null property explicitly.
- The `empty` is for testing optional property is not sent. (The service returns null property implicitly)

Since they do not be related to inheritance and I find the `empty` has just been covered in `property-optional`, I remove them from this PR. And trying to create new PR to add `null` test in `property-optional`.